src/systemd-sonic-generator/systemd-sonic-generator.cpp

```diff
@@ -385,6 +388,56 @@ static void replace_multi_inst_dep(const char *src) {
     rename(tmp_file_path, src);
 }
 
+static void update_environment(const std::string &src)
+{
+    std::ifstream src_file(src);
+    std::ofstream tmp_file(src + ".tmp");
+    bool has_service_section = false;
+    std::string line;
+
+    // locate the [Service] section
+    while (std::getline(src_file, line)) {
+        tmp_file << line << std::endl;
+        if (line.find("[Service]") != std::string::npos) {
+            has_service_section = true;
+            break;
+        }
+    }
+
+    if (!has_service_section) {
+        tmp_file << "[Service]" << std::endl;
+    }
+
+    std::unordered_map<std::string, std::string> env_vars;
+    env_vars["IS_DPU_DEVICE"] = (smart_switch_dpu ? "true" : "false");
+    env_vars["NUM_DPU"] = std::to_string(num_dpus);
+
+    for (const auto& [key, value] : env_vars) {
+        tmp_file << "Environment=\"" << key << "=" << value << "\"" << std::endl;
```

Comment:
Do we need to add Environment= option if the file doesn't have`[Service]` section in the file

Reply:
I worry that the `Service` section might be introduced by other functions or modules in the future. And I don't see any side-effects if I define some environment variables at this point.

Reply:
Please make sure, systemd won't throw an error if we add Environment= values after the last section without [Service]

Reply:
If there is no Service section, I will add it.

![image](https://github.com/user-attachments/assets/40ea9b1f-042d-4b8d-b1a5-45d19290a583)

Reply:
Sounds good to me